contracts/dynamic-contracts/extension/PermissionsSig.sol

```diff
@@ -0,0 +1,310 @@
+    }
+
+    /**
+     *  @notice         Returns the admin role that controls the specified role.
+     *  @dev            See {grantRole} and {revokeRole}.
+     *                  To change a role's admin, use {_setRoleAdmin}.
+     *
+     *  @param role     keccak256 hash of the role. e.g. keccak256("TRANSFER_ROLE")
+     */
+    function getRoleAdmin(bytes32 role) external view override returns (bytes32) {
+        PermissionsSigStorage.Data storage data = PermissionsSigStorage.permissionsSigStorage();
+        return data._getRoleAdmin[role];
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                            External functions
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     *  @notice         Grants a role to an account, if not previously granted.
+     *  @dev            Caller must have admin role for the `role`.
+     *                  Emits {RoleGranted Event}.
+     *
+     *  @param req     The request body to grant a role to an account.
+     *  @param signature  Signature of a party authorized to grant role.
+     */
+    function grantRole(RoleRequest calldata req, bytes calldata signature) public virtual override {
+        require(req.action == RoleAction.Grant, "Invalid action");
+        _processRoleRequest(req, signature);
+        _setupRole(req.role, req.target);
+    }
+
+    /**
+     *  @notice         Revokes role from an account.
+     *  @dev            Caller must have admin role for the `role`.
+     *                  Emits {RoleRevoked Event}.
+     *
+     *  @param req     The request body to revoke a role from an account.
+     *  @param signature  Signature of a party authorized to revoke role.
+     */
+    function revokeRole(RoleRequest calldata req, bytes calldata signature) public virtual override {
+        require(req.action == RoleAction.Revoke, "Invalid action");
+        _processRoleRequest(req, signature);
+        _revokeRole(req.role, req.target);
+    }
+
+    /**
+     *  @notice         Revokes role from the account.
+     *  @dev            Caller must have the `role`, with caller being the same as `account`.
+     *                  Emits {RoleRevoked Event}.
+     *
+     *  @param req     The request body to renounce a role.
+     *  @param signature  Signature of a party authorized to renounce role.
+     */
+    function renounceRole(RoleRequest calldata req, bytes calldata signature) public virtual override {
+        require(req.action == RoleAction.Renounce, "Invalid action");
+        _processRoleRequest(req, signature);
+        _revokeRole(req.role, req.target);
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                            Internal functions
+    //////////////////////////////////////////////////////////////*/
+
+    /// @dev Sets `adminRole` as `role`'s admin role.
+    function _setRoleAdmin(bytes32 role, bytes32 adminRole) internal virtual {
```

Comment:
when is this used?

Reply:
We don't use this currently. I left it in there as it is a part of `Permissions` in general.

Example usage of the function: your contract already has `DEFAULT_ADMIN_ROLE` (which is the role admin of all roles by default) and `MINTER_ROLE`. You can create a `MINTER_ADMIN_ROLE` and call `_setRoleAdmin` to ensure only `MINTER_ADMIN_ROLE` role holders (and not `DEFAULT_ADMIN_ROLE` holders) can grant / revoke `MINTER_ROLE`.